generators/server/index.js

```diff
@@ -936,7 +954,7 @@ module.exports = JhipsterServerGenerator.extend({
             this.template(SERVER_MAIN_SRC_DIR + 'package/security/_SecurityUtils.java', javaDir + 'security/SecurityUtils.java', this, {});
             this.template(SERVER_MAIN_SRC_DIR + 'package/security/_AuthoritiesConstants.java', javaDir + 'security/AuthoritiesConstants.java', this, {});
 
-            if (this.authenticationType === 'jwt') {
+            if (this.authenticationType == 'jwt' && this.applicationType == 'monolith') {
```

Comment:
check should be reversed `this.applicationType == 'monolith' && this.authenticationType == 'jwt'`